docs/proposals/20200220-cluster-resource-set.md

```diff
@@ -85,6 +88,9 @@ As someone creating multiple clusters, I want some/all my clusters to have a CNI
 As someone creating multiple clusters, I want some/all my clusters to have a StorageClass installed automatically, so I don't have to manually repeat the installation for each new cluster.
 
 #### Story 3
+As someone managing multiple clusters, I want to be able to update the CPI and CSI by just updating the the `ConfigMap`'s and `Secret`'s in the management cluster, so I don't have to manually repeat the apply for each targeted cluster.
```

Comment:
Q: Am I wrong or CPI/CSI version is often linked to the Kubernetes version, so isn't it a risk to have this addon automatically updated in all the workload clusters, no matter the Kubernetes version currently in use by each cluster?

Reply:
You are right. The user would need to make sure all kubernetes cluster are upgraded at once.

Which makes me think, maybe I should rephrase this to describe an scenario where each cluster has its own CRS for CPI/CSI? That might more realistic.

Reply:
> a scenario where each cluster has its own CRS for CPI/CSI

IMO having a single CRS for each cluster defeats the goal of CRS, which is designed to work with many clusters; I don't think we should suggest this to users (it is more a workaround than a feauture)
TL;DR; I don't think CRS is the best tool for this use case given the limitations of the original design

Reply:
I agree, it's not the best tool. I think of this as a small and "cheap" improvement that allows us to solve a present problem, even though not in the ideal way, until we have a better tool available. 

Would you then suggest to remove this user story completely?

Reply:
+1 to remove the story or to replace CPI/CSI with some other example.

Reply:
Removed